frontend/src/metabase/dashboard/components/DashboardTabs/useDashboardTabs.ts

```diff
@@ -7,31 +8,50 @@ import {
   deleteTab as deleteTabAction,
   initTabs,
   selectTab as selectTabAction,
+  undoDeleteTab,
 } from "metabase/dashboard/actions";
-import { SelectedTabId, State } from "metabase-types/store";
-import { getDashboardId } from "metabase/dashboard/selectors";
+import { SelectedTabId } from "metabase-types/store";
+import { getDashboardId, getSelectedTabId } from "metabase/dashboard/selectors";
+import { addUndo } from "metabase/redux/undo";
 
-export function getSelectedTabId(state: State) {
-  return state.dashboard.selectedTabId;
-}
+let tabDeletionId = 1;
 
 export function useDashboardTabs() {
   const dispatch = useDispatch();
   const dashboardId = useSelector(getDashboardId);
   const tabs = useSelector(state =>
     dashboardId
-      ? state.dashboard.dashboards[dashboardId].ordered_tabs ?? []
+      ? state.dashboard.dashboards[dashboardId].ordered_tabs?.filter(
+          tab => !tab.isRemoved,
+        ) ?? []
       : [],
   );
   const selectedTabId = useSelector(getSelectedTabId);
 
   useMount(() => dispatch(initTabs()));
 
+  const deleteTab = (tabId: SelectedTabId) => {
```

Comment:
I prefer all the logic to be in the reducer as is for all other tab actions, however in this case it's necessary to put some logic here in order to dispatch the `addUndo` action.